src/renderer/renderer.ts

```diff
@@ -856,9 +857,35 @@ export class SDFGRenderer extends EventEmitter {
         else
             this.bgcolor = window.getComputedStyle(this.canvas).backgroundColor;
 
+
         // Create the initial SDFG layout
         // Loading animation already started in the file_read_complete function in sdfv.ts
         // to also include the JSON parsing step.
+        
+        // Initial layout fully collapsed
+        this.for_all_sdfg_elements(
+            (_t: SDFGElementGroup, _d: any, obj: any) => {
+                if ('is_collapsed' in obj.attributes &&
+                    !obj.type.endsWith('Exit'))
+                    obj.attributes.is_collapsed = true;
+            }
+        );
+        this.emit('collapse_state_changed', true, true);
+        this.relayout();
```

Comment:
Why relayout twice?

Reply:
Oh yes this was a quick fix because there was an error without the this.relayout() in-between the fully collapse and expand. I just took the existing collapse and expand code from the renderer. Relayouting after fully collapsing here only takes a millisecond, so if it is really necessary it's only a small overhead. I will check again though!